### DISCUSSION TITLE
Sample Uno App not working

### DISCUSSION THREAD

**Discussion by yogeshdTE** [unoplatform/uno#11833] [General]
I have a ARM32 based system (AM335X SoC) with customized Yocto Linux build with FrameBuffer support and trying to evaluate Uno Platform to run existing WPF application on the same. When tried running ListViewSample (which demonstrates basic WPF binding with ListView) on the system by adding Skia.Linux.FrameBuffer header project, the white blank screen appeared. 
1. Hence set the height and width to the WPF Page defined in the shared project and set the Arial font with size 30 explicitly to the WPF page. With this, "My Connection" text appeared on the screen as a header with a ListView with different gradient color for alternate rows but no text appeared in the ListView.
2. Then explicitly specified Font size for different columns in the ListView data template, but no text displayed in the ListView. Also added few log lines in MainPage.xaml.cs to check data binding and those lines are printed correctly.
3. Gone through few discussions (https://github.com/unoplatform/uno/discussions/11800, https://github.com/unoplatform/uno/discussions/9473, https://github.com/unoplatform/uno/discussions/8737) and installed few fonts making sure that all fonts are installed, but still not able to display any text on ListView.

![ListViewSampleOutput](https://user-images.githubusercontent.com/98949235/228442770-e9222849-ced7-4e3d-b448-5d236bb2d1ab.jpeg)

I made sure all latest NuGet packages are used. So, Can someone help me to find out the issue here?

Also, few observations here:
1. after invoking a ListViewSample from console, it takes ~10 secs to load the application. Hence, as suggested in one of the above discussions, tried to publish the application with p:PublishReadyToRun=false option, which is failing with the error  
_C:\Program Files\dotnet\sdk\7.0.100\Sdks\Microsoft.NET.Sdk\targets\Microsoft.NET.CrossGen.targets(472,28): error MSB4094: "obj\Release\net6.0\linux-arm\linked\SkiaSharp.Views.Windows.dll;obj\Release\net6.0\linux-arm\linked\SkiaSharp.Views.Windows.dll;obj\Release\net6.0\linux-arm\linked\SkiaSharp.Views.Windows.dll;obj\Release\net6.0\linux-arm\linked\SkiaSharp.Views.Windows.dll" is an invalid value for the "CompilationEntry" parameter of the "RunReadyToRunCompiler" task. Multiple items cannot be passed into a parameter of type "Microsoft.Build.Framework.ITaskItem"._
Here is the publish command used to build the app: _dotnet publish -c Release -o publish -r linux-arm -p:PublishReadyToRun=false -p:PublishSingleFile=true -p:PublishTrimmed=true --self-contained true -p:IncludeNativeLibrariesForSelfExtract=true_
2. Also, After selecting one of the rows of the ListView using mouse, the row is selected i.e. blue background appears on the row after ~ 2 mins. Any specific reason for this behavior and any solution?

Can someone help me in resolving above issues? Let me know in case of any details.

**Comment by jeromelaban** [comment]
Thanks for the report.

- For the fonts, some distros don't include default Microsoft related fonts, make sure to [take a look at this section](https://platform.uno/docs/articles/get-started-with-linux.html?tabs=ubuntu1804#setting-up-for-linux). Troubleshooting tips may also be found in https://github.com/unoplatform/uno/issues/6441.

- The ReadyToRun issue is unknown. You may want to try with net7.0 and see if it helps. Otherwise, open an issue with [build logs](https://platform.uno/docs/articles/uno-builds-troubleshooting.html). You may get better results with Native AOT as well.

- For the 2 minutes to show a selection, this is unknown as well and seems particularly long even for a low-powered device like this. If you can provide a repro scenario it may help troubleshooting.

**Reply by yogeshdTE** [reply]
Hi jeromelaban,
Regarding fonts, I am able to resolve the problem in the above sample by adding support for Material theme for framebuffer project header. Original example does not have framebuffer project header; I added it explicitly (from UnoGoodReads example) but somehow missed the Material support and the text was not displaying. 

I also tried ReadyToRun option with .Net 7, but still there is an issue in publishing.

Regarding show a selection, we are still working on few activities to evaluate Uno Platform on this front. I will keep you posted our outcome.

Thanks.
